src/sql/workbench/contrib/notebook/test/browser/notebookActions.test.ts

```diff
@@ -3,18 +3,104 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import * as TypeMoq from 'typemoq';
 import * as assert from 'assert';
-
-import { AddCellAction, ClearAllOutputsAction, CollapseCellsAction, TrustedAction, RunAllCellsAction, NewNotebookAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
-import { CellType } from 'sql/workbench/services/notebook/common/contracts';
-import { INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import * as azdata from 'azdata';
+import * as sinon from 'sinon';
+import { TestConfigurationService } from 'sql/platform/connection/test/common/testConfigurationService';
+import { AddCellAction, ClearAllOutputsAction, CollapseCellsAction, KernelsDropdown, msgChanging, NewNotebookAction, noKernelName, RunAllCellsAction, TrustedAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
+import { ClientSessionStub as stubs_ClientSessionStub, ContextViewProviderStub, NotebookComponentStub, NotebookModelStub as stubs_NotebookModelStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { NotebookEditorStub } from 'sql/workbench/contrib/notebook/test/testCommon';
 import { ICellModel, INotebookModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { IStandardKernelWithProvider } from 'sql/workbench/services/notebook/browser/models/notebookUtils';
+import { INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import { CellType } from 'sql/workbench/services/notebook/common/contracts';
+import * as TypeMoq from 'typemoq';
+import { Emitter } from 'vs/base/common/event';
+import { TestCommandService } from 'vs/editor/test/browser/editorTestServices';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { IConfigurationChangeEvent, IConfigurationOverrides, IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { TestInstantiationService } from 'vs/platform/instantiation/test/common/instantiationServiceMock';
 import { INotificationService } from 'vs/platform/notification/common/notification';
 import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
-import { ICommandService } from 'vs/platform/commands/common/commands';
-import { TestCommandService } from 'vs/editor/test/browser/editorTestServices';
-import { NotebookComponentStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { workbenchInstantiationService } from 'vs/workbench/test/browser/workbenchTestServices';
+
+class ClientSessionStub extends stubs_ClientSessionStub {
```

Comment:
>ClientSessionStub [](start = 6, length = 17)

can we substitute sinon.stub for the specific methods instead of creating the same class?

Reply:
can be done but the same stubs will have to be done in several tests.